File: rb/112.rb
def bouncy? n
  c = n.to_s.chars.to_a
  c != (s = c.sort) && c != s.reverse
end

b, n = 0, 0
until b / n.to_f == 0.99
  n += 1
  b += bouncy?(n) ? 1 : 0
end

p n

File: js/112.js
function bouncy(n) {
    var c = n.toString().split(''),
        s = c.slice(0).sort();
    return c.toString() != s.toString() && c.toString() != s.reverse().toString();
}

var b = n = 0;
while (b / n != 0.99)
    b += bouncy(n += 1);

console.log(n);
